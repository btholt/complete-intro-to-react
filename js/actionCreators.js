import { SET_SEARCH_TERM } from './actions';

export function setSearchTerm(searchTerm) {
  return { type: SET_SEARCH_TERM, payload: searchTerm };
}
