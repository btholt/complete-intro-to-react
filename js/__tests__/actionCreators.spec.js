// @flow

import { setSearchTerm, addAPIData } from '../actionCreators';

test('setSearchTerm', () => {
  expect(setSearchTerm('New York')).toMatchSnapshot();
});

test('addAPIData', () => {
  expect(
    addAPIData({
      title: 'Stranger Things',
      year: '2016â€“',
      description:
        'When a young boy disappears, his mother, a police chief, and his friends must confront terrifying forces in order to get him back.',
      poster: 'st.jpg',
      imdbID: 'tt4574334',
      trailer: '9Egf5U8xLo8',
      rating: '8.6'
    })
  ).toMatchSnapshot();
});
