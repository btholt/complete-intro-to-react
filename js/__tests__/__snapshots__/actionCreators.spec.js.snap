// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addAPIData 1`] = `
Object {
  "payload": Object {
    "description": "When a young boy disappears, his mother, a police chief, and his friends must confront terrifying forces in order to get him back.",
    "imdbID": "tt4574334",
    "poster": "st.jpg",
    "rating": "8.6",
    "title": "Stranger Things",
    "trailer": "9Egf5U8xLo8",
    "year": "2016â€“",
  },
  "type": "ADD_API_DATA",
}
`;

exports[`setSearchTerm 1`] = `
Object {
  "payload": "New York",
  "type": "SET_SEARCH_TERM",
}
`;
